LogicMigrator/Migrator.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using MMRando.Extensions;
+using MMRando.Utils;
+using static MMRando.MainRandomizerForm;
+
+namespace MMRando.LogicMigrator
+{
+    public static partial class Migrator
+    {
+        public const int CurrentVersion = 2;
+
+        public static string ApplyMigrations(string logic)
+        {
+            var lines = logic.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None).ToList();
+
+            if (GetVersion(lines) < 0)
+            {
+                AddVersionNumber(lines);
+            }
+
+            if (GetVersion(lines) < 1)
+            {
+                AddItemNames(lines);
+            }
+
+            if (GetVersion(lines) < 2)
+            {
+                AddMoonItems(lines);
+            }
+
+            return string.Join("\r\n", lines);
+        }
+
+        private static int GetVersion(List<string> lines)
+        {
+            if (!lines[0].StartsWith("-version"))
+            {
+                return -1;
+            }
+            return int.Parse(lines[0].Split(' ')[1]);
+        }
+
+        private static void AddVersionNumber(List<string> lines)
+        {
+            lines.Insert(0, "-version 0");
+        }
+
+        private static void AddItemNames(List<string> lines)
+        {
+            if (lines[1] == "- Deku Mask")
+            {
+                lines[0] = "-version 1";
+                return;
+            }
+            lines.RemoveAll(line => line.StartsWith("-"));
+            var itemNames = new string[] {"Deku Mask", "Hero's Bow", "Fire Arrow", "Ice Arrow", "Light Arrow", "Bomb Bag (20)", "Magic Bean",
```

Comment:
Isn't this already defined in another location?

Reply:
It is, but there needs to be a copy here in order for the logic migration to work regardless of the codebase. I.e. if we rearrange the items list, this list will need to remain as it is.